client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,392 @@
+[source,java]
+----
+Builder builderWithSugar = new Builder(Operation.Type.QUERY)
+       .addRootField("allHeroes", fields(field("name")));
+----
+
+The static methods to obtain this lighter syntax are described in the <<appendix_static_methods>>.
+
+=== Building a more complex query
+
+In this example, we want to get all heroes from a given city, including the name of the teams they belong to.
+The GraphQL query that fits this need is:
+
+[source,json]
+----
+query {
+  allHeroesIn(city: "New York, NY") {
+	name
+	currentLocation
+	teamAffiliations {
+  		name
+	}
+  }
+----
+The way to build this query with the API is:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.QUERY)
+       .addRootField(new Field("allHeroesIn",
+               new HashSet<Argument>(asList(
+                       new Argument("city", "New York, NY")
+               )),
+               new ArrayList<>(asList(
+                       new Field("name"),
+                       new Field("currentLocation"),
+                       new Field("teamAffiliations", new ArrayList<>(asList(
+                               new Field("name")))
+                       )))));
+----
+The same query with the ligther syntax:
+[source,java]
+----
+Builder builderWithSugar = new Builder(Operation.Type.QUERY)
+       .addRootField("allHeroesIn",
+               args(
+                       arg("city", "New York, NY")
+               ),
+               fields(
+                       field("name"),
+                       field("currentLocation"),
+                       field("teamAffiliations",
+                               fields(
+                                       field("name")
+                               )
+                       )
+               )
+       );
+----
+
+=== Adding attributes to fields
+
+In the previous example, there is just one argument at the root level. What if we want to set arguments to fields?
+Let’s imagine we want to add 2 arguments to the “currentLocation” field:
+[source, java]
+----
+Builder builder = new Builder(Operation.Type.QUERY)
+       .addRootField("allHeroesIn",
+               new HashSet<Argument>(asList(
+                       new Argument("city", "New York, NY")
+               )),
+               new HashSet<Field>(asList(
+                       new Field("name"),
+                       new Field("currentLocation", new HashSet<Argument>(asList(
+                               new Argument("foo", "bar"),
+                               new Argument("zoo", 123)
+                       ))),
+                       new Field("teamAffiliations", new HashSet<Field>(asList(
+                               new Field("name")))
+                       ))))
+       .addRootField("allHeroes", new HashSet<Field>(asList(new Field("name"))));
+
+----
+
+With the light API:
+[source,java]
+----
+Builder builderWithSugar = new Builder(Operation.Type.QUERY)
+       .addRootField("allHeroesIn",
+               asSet(
```

Comment:
Is `asSet` our own helper method? In the example above, there was `fields` and `args`.